ntoskrnl/ex/time.c

```diff
@@ -468,6 +527,15 @@ NtSetSystemTime(IN PLARGE_INTEGER SystemTime,
         _SEH2_END;
     }
 
+    /* Read time zone information from the registry */
+    Status = RtlQueryTimeZoneInformation(&TimeZoneInformation);
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT1("RtlQueryTimeZoneInformation() failed (Status 0x%08lx)\n", Status);
+    }
+
+    ExpSetTimeZoneInformation(&TimeZoneInformation);
```

Comment:
```suggestion
        DPRINT1("RtlQueryTimeZoneInformation() failed (Status 0x%08lx)\n", Status);
    else
        Status = ExpSetTimeZoneInformation(&TimeZoneInformation);
```

Reply:
I think that we want to do this even if we receive a failure code from RtlQueryTimeZoneInformation. It's just the best that we can do here. Thanks.

Reply:
Using an uninitialized `TimeZoneInformation` doesn't look good to me.

Reply:
@SergeGautherie, We set the TimeZoneInformation values to all zeroes here:

RTL_TIME_ZONE_INFORMATION TimeZoneInformation = {0};

Then we make the call to RtlQueryTimieZoneInformation like this:

Status = RtlQueryTimeZoneInformation(&TimeZoneInformation);

RtlQueryTimeZoneInformation tries to fill this struct with values from the registry.
If it fails, then the original zeroes just remain in place.
Proceeding with zeroes in the TimeZoneInformation structure basically just leaves things alone.
So basically the DaylightDate, DaylightBias, StandardDate, and StandardBias members are all zero.
So changes occur in the code below and no harm is done.